micrometer-core/src/test/java/io/micrometer/core/instrument/distribution/DistributionStatisticConfigExtensibilityTest.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2023 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.distribution;
+
+import io.micrometer.common.lang.Nullable;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class DistributionStatisticConfigExtensibilityTest {
+
+    static class SubDistributionStatisticConfig extends DistributionStatisticConfig {
+
+        public static final SubDistributionStatisticConfig NONE = subBuilder().build();
+
+        @Nullable
+        protected String extraParam;
+
+        @Nullable
+        public String getExtraParam() {
+            return extraParam;
+        }
+
+        SubDistributionStatisticConfig() {
+            super();
+        }
+
+        SubDistributionStatisticConfig(DistributionStatisticConfig original) {
+            super(original);
+            if (original instanceof SubDistributionStatisticConfig) {
+                extraParam = ((SubDistributionStatisticConfig) original).extraParam;
+            }
+        }
+
+        @Override
+        public SubDistributionStatisticConfig merge(DistributionStatisticConfig parent) {
+            DistributionStatisticConfig mergeBySuper = super.merge(parent);
+            SubDistributionStatisticConfig sub = new SubDistributionStatisticConfig(mergeBySuper);
+            if (extraParam == null && parent instanceof SubDistributionStatisticConfig) {
+                sub.extraParam = ((SubDistributionStatisticConfig) parent).extraParam;
+            }
+            else {
+                sub.extraParam = extraParam;
+            }
+            return sub;
+        }
+
+        public static SubBuilder subBuilder() {
+            return new SubBuilder(new SubDistributionStatisticConfig());
+        }
+
+        public static class SubBuilder extends Builder {
```

Comment:
Is this why you need to receive a `DistributionStatisticConfig` in `DistributionStatisticConfig.Builder`?

I don't think `SubDistributionStatisticConfig.Builder` should extend `DistributionStatisticConfig.Builder`.

Reply:
Yes. Which way would you recommend?